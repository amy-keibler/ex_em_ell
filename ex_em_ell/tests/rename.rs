#[derive(Debug, PartialEq, ex_em_ell::FromXmlDocument, ex_em_ell::ToXmlDocument)]
struct Example {
    // Support Acronym Case conventions without forcing the Rust variable to be field_u_r_l
    #[ex_em_ell(rename = "fieldURL")]
    field_url: String,

    #[ex_em_ell(rename = "childURL")]
    child_url: ExampleChild,
}

#[derive(Debug, PartialEq, ex_em_ell::FromXmlElement, ex_em_ell::ToXmlElement)]
struct ExampleChild {
    #[ex_em_ell(rename = "fieldURL")]
    field_url: String,
}

#[test]
fn test_example_xmls() {
    insta::glob!("data/rename/rename_*.xml", |path| {
        let file = std::fs::File::open(path).expect(&format!("Failed to read file: {path:?}"));
        let example: Example = ex_em_ell::from_reader(&file)
            .expect(&format!("Failed to parse the XML file: {path:?}"));

        let round_trip = ex_em_ell::to_string_pretty(&example).expect("Failed to output XML");
        insta::assert_snapshot!(round_trip);
    });
}
