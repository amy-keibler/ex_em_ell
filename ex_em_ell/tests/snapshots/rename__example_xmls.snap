---
source: ex_em_ell/tests/rename.rs
expression: round_trip
input_file: ex_em_ell/tests/data/rename/rename_simple.xml
---
<?xml version="1.0" encoding="utf-8"?>
<example>
  <fieldURL>value</fieldURL>
  <childURL>
    <fieldURL>value</fieldURL>
  </childURL>
</example>
